internal/app/http/service.go

```diff
@@ -0,0 +1,213 @@
+	rm, err := dhs.getRemoteManager()
+	if err != nil {
+		return nil, err
+	}
+
+	c := commands.NewListCommand(rm)
+	toc, err := c.List(remotes.EmptySpec, search)
+	if err != nil {
+		return nil, err
+	}
+
+	return &toc, nil
+}
+
+func (dhs *defaultHandlerService) ListAuthors(ctx context.Context, search *model.SearchParams) ([]string, error) {
+	authors := []string{}
+
+	toc, err := dhs.ListInventory(ctx, search)
+	if err != nil {
+		return authors, err
+	}
+
+	check := map[string]bool{}
+	for _, v := range toc.Entries {
+		if _, ok := check[v.Author.Name]; !ok {
+			check[v.Author.Name] = true
+			authors = append(authors, v.Author.Name)
+		}
+	}
+	sort.Strings(authors)
+	return authors, nil
+}
+
+func (dhs *defaultHandlerService) ListManufacturers(ctx context.Context, search *model.SearchParams) ([]string, error) {
+	mans := []string{}
+
+	toc, err := dhs.ListInventory(ctx, search)
+	if err != nil {
+		return mans, err
+	}
+
+	check := map[string]bool{}
+	for _, v := range toc.Entries {
+		if _, ok := check[v.Manufacturer.Name]; !ok {
+			check[v.Manufacturer.Name] = true
+			mans = append(mans, v.Manufacturer.Name)
+		}
+	}
+	sort.Strings(mans)
+	return mans, nil
+}
+
+func (dhs *defaultHandlerService) ListMpns(ctx context.Context, search *model.SearchParams) ([]string, error) {
+	mpns := []string{}
+
+	toc, err := dhs.ListInventory(ctx, search)
+	if err != nil {
+		return mpns, err
+	}
+
+	check := map[string]bool{}
+	for _, v := range toc.Entries {
+		if _, ok := check[v.Mpn]; !ok {
+			check[v.Mpn] = true
+			mpns = append(mpns, v.Mpn)
+		}
+	}
+	sort.Strings(mpns)
+	return mpns, nil
+}
+
+func (dhs *defaultHandlerService) FindInventoryEntry(ctx context.Context, name string) (*model.FoundEntry, error) {
+	//todo: check if name is valid format
+	toc, err := dhs.ListInventory(ctx, &model.SearchParams{Name: name})
+	if err != nil {
+		return nil, err
+	}
+	if len(toc.Entries) != 1 {
+		return nil, NewNotFoundError(nil, "Inventory with name %s not found", name)
+	}
+	return &toc.Entries[0], nil
+}
+
+func (dhs *defaultHandlerService) FetchThingModel(ctx context.Context, tmID string) ([]byte, error) {
+	_, err := model.ParseTMID(tmID, true)
+	if errors.Is(err, model.ErrInvalidId) {
+		return nil, NewBadRequestError(err, "Invalid parameter: %s", tmID)
+	} else if err != nil {
+		return nil, err
+	}
+
+	rm, err := dhs.getRemoteManager()
+	if err != nil {
+		return nil, err
+	}
+
+	_, data, err := commands.NewFetchCommand(rm).FetchByTMID(remotes.EmptySpec, tmID)
+	if errors.Is(err, commands.ErrTmNotFound) {
+		return nil, NewNotFoundError(err, "File does not exist")
+	} else if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+func (dhs *defaultHandlerService) PushThingModel(ctx context.Context, file []byte) (string, error) {
+	rm, err := dhs.getRemoteManager()
+	if err != nil {
+		return "", err
+	}
+
+	remoteSpec := dhs.pushRemote
+	if remoteSpec.IsEmpty() {
```

Comment:
What's the problem with the push target spec being empty? The validity of the spec is already [verified on startup](https://github.com/web-of-things-open-source/tm-catalog-cli/blob/test/api-handler/internal/app/cli/serve.go#L21) by the time we get here.

Reply:
I removed this empty spec validation, as it was really unnecessary.